source/Application.cpp

```diff
@@ -0,0 +1,147 @@
+        int optionPadding = 8;
+
+        tgui::Panel::Ptr panel = tgui::Panel::create();
+        panel->setSize({ "100%", (options.size() + 1) * optionHeight });
+        panel->setAutoLayout(tgui::AutoLayout::Top);
+
+        tgui::Label::Ptr labelName = tgui::Label::create(name);
+        labelName->setSize({ "100% - " + tgui::String::fromNumber(2 * optionPadding), optionHeight - 2 * optionPadding });
+        labelName->setPosition({ optionPadding, optionPadding });
+        labelName->getRenderer()->setTextStyle(tgui::TextStyle::Bold);
+        labelName->setScrollbarPolicy(tgui::Scrollbar::Policy::Never);
+        panel->add(labelName);
+
+        for (ushort i = 0; i < options.size(); i += 1) {
+            tgui::Group::Ptr groupOption = tgui::Group::create();
+            groupOption->setPosition({ 0, (i + 1) * optionHeight });
+            groupOption->setSize({ "100%", optionHeight });
+            panel->add(groupOption);
+
+            tgui::Picture::Ptr pictureOption = tgui::Picture::create("assets/images/button-red.png", true);
+            pictureOption->setPosition({ optionPadding, optionPadding * 1.5f });
+            pictureOption->setSize({ 8, 8 });
+            groupOption->add(pictureOption);
+
+            tgui::Label::Ptr labelOption = tgui::Label::create(options[i]);
+            labelOption->setPosition({ 24, optionPadding });
+            labelOption->setSize({ "100% - " + tgui::String::fromNumber(optionPadding + 24), optionHeight - 2 * optionPadding });
+            labelOption->setScrollbarPolicy(tgui::Scrollbar::Policy::Never);
+            groupOption->add(labelOption);
+
+            tgui::ClickableWidget::Ptr buttonOption = tgui::ClickableWidget::create();
+            buttonOption->setPosition({ 0, 0 });
+            buttonOption->setSize({ "100%", "100%" });
+            buttonOption->onClick([=]() {
+                std::cout << "Option " << options[i] << " pressed" << std::endl;
+            });
+            groupOption->add(buttonOption);
+        }
+
+        return panel;
+    }
+
+    tgui::Button::Ptr Application::buildSiderButton(tgui::Panel::Ptr sider)
+    {
+        tgui::Button::Ptr button = tgui::Button::create("X");
+        button->setSize({ "20%", "5%" });
+        button->setPosition({ "0", "95%" });
+        button->onPress([=]() {
+            if (sider->isVisible()) {
+                std::cout << "HIDE" << std::endl;
+                sider->moveWithAnimation({ "-width", "0" }, sf::milliseconds(ANIMATION_DURATION));
+                button->resizeWithAnimation({ "64", button->getSize().y }, sf::milliseconds(ANIMATION_DURATION));
+
+                /*
+                unsigned int signalId = sider->onAnimationFinish([=]() {
+                    sider->setVisible(false);
+                    sider->onAnimationFinish.disconnect(signalId);
+                    std::cout << "Animation finished with id " << signalId << std::endl;
+                });
+                /*/
```

Comment:
Let's clean the code `std::cout` and commented-out code if they serve no purpose